pkgs/by-name/sp/spotify-backup/package.nix

```
@@ -14,14 +15,12 @@ python3.pkgs.buildPythonPackage {
     hash = "sha256-+z9IWgtc71GPPLDqNU4PXFDyD5Dczp3Bbrwzy/DaIts=";
   };
 
-  preBuild = ''
-    cat > setup.py << EOF
-from setuptools import setup
-setup(scripts=['spotify-backup.py'])
-EOF'';
+  propagatedBuildInputs = [
+    python3
+  ];
 
-  postInstall = ''
-    mv -v $out/bin/spotify-backup.py $out/bin/spotify-backup
+  installPhase = ''
+    install -Dm755 spotify-backup.py $out/bin/spotify-backup
   '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm755 spotify-backup.py $out/bin/spotify-backup
    runHook postInstall
  '';
```

when overriding phases, you need to run their hooks manually.

---

alternatively, you could move this back to postInstall if you want to keep the nix expression small.